doc/dev_guide/c_style_guide_new.rst

```diff
@@ -0,0 +1,920 @@
+        typedef unsigned long myflags_t;
+
+    but if there is a clear reason for why it under certain circumstances
+    might be an ``unsigned int`` and under other configurations might be
+    ``unsigned long``, then by all means go ahead and use a typedef.
+
+#.  when you use sparse to literally create a **new** type for
+    type-checking.
+
+#.  New types which are identical to standard C99 types, in certain
+    exceptional circumstances.
+
+    Although it would only take a short amount of time for the eyes and
+    brain to become accustomed to the standard types like ``uint32_t``,
+    some people object to their use anyway.
+
+    Therefore, the Linux-specific ``u8/u16/u32/u64`` types and their
+    signed equivalents which are identical to standard types are
+    permitted -- although they are not mandatory in new code of your
+    own.
+
+    When editing existing code which already uses one or the other set
+    of types, you should conform to the existing choices in that code.
+
+Maybe there are other cases too, but the rule should basically be to NEVER
+EVER use a typedef unless you can clearly match one of those rules.
+
+In general, a pointer, or a struct that has elements that can reasonably
+be directly accessed should **never** be a typedef.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Chapter 6: Functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Functions should be short and sweet, and do just one thing. They should
+fit on one or two screenfuls of text (the ISO/ANSI screen size is 80x24,
+as we all know), and do one thing and do that well.
+
+The maximum length of a function is inversely proportional to the
+complexity and indentation level of that function. So, if you have a
+conceptually simple function that is just one long (but simple)
+case-statement, where you have to do lots of small things for a lot of
+different cases, it's OK to have a longer function.
+
+However, if you have a complex function, and you suspect that a
+less-than-gifted first-year high-school student might not even
+understand what the function is all about, you should adhere to the
+maximum limits all the more closely. Use helper functions with
+descriptive names (you can ask the compiler to in-line them if you think
+it's performance-critical, and it will probably do a better job of it
+than you would have done).
+
+Another measure of the function is the number of local variables. They
+shouldn't exceed 5-10, or you're doing something wrong. Re-think the
+function, and split it into smaller pieces. A human brain can
+generally easily keep track of about 7 different things, anything more
+and it gets confused. You know you're brilliant, but maybe you'd like
+to understand what you did 2 weeks from now.
+
+In function prototypes, include parameter names with their data types.
+Although this is not required by the C language, it is preferred in Linux
```

Comment:
"in Linux" -> "in Tarantool". Might be good to find all the usages of word "Linux" and replace with "Tarantool" where appropriate.